### PR TITLE
Hand through intensities into the generated PLY - if they exist.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

File: cartographer/io/ply_writing_points_processor.cc [hunk 1]
```diff
@@ -61,6 +64,11 @@ void WriteBinaryPlyPointCoordinate(const Eigen::Vector3f& point,
   CHECK(file_writer->Write(buffer, 12));
 }
 
+void WriteBinaryIntensity(const float intensity,
+                          FileWriter* const file_writer) {
+  CHECK(file_writer->Write(&intensity, sizeof(float)));
```
Comment (wohe) [inline cartographer/io/ply_writing_points_processor.cc]: Does this compile? It looks like it is missing a `reinterpret_cast<const char*>`.

Comment (SirVer) [reply cartographer/io/ply_writing_points_processor.cc]: does now.

File: cartographer/io/ply_writing_points_processor.cc [hunk 2]
```diff
@@ -61,6 +64,12 @@ void WriteBinaryPlyPointCoordinate(const Eigen::Vector3f& point,
   CHECK(file_writer->Write(buffer, 12));
 }
 
+void WriteBinaryIntensity(const float intensity,
+                          FileWriter* const file_writer) {
+  CHECK(file_writer->Write(reinterpret_cast<const char*>(&intensity),
```
Comment (wohe) [inline cartographer/io/ply_writing_points_processor.cc]: Do you want to add a `TODO`about the endianness bug?

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Pull request merged**